src/ralph/accounts/ldap.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# max-complexity=25
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+
+from dj.choices import Country
+
+
+logger = logging.getLogger(__name__)
+try:
+    from django_auth_ldap.config import ActiveDirectoryGroupType
+    from django_auth_ldap.backend import (
+        populate_user,
+        LDAPSettings,
+    )
+except ImportError:
+    logger.debug("django_auth_ldap package not provided")
+else:
+    from django.dispatch import receiver
+    from django.conf import settings
+    from django.utils.encoding import force_unicode
+
+    # Add default value to LDAPSetting dict. It will be replaced by
+    # django_auth_ldap with value in settings.py and visible for each
+    # ldap_user.
+    LDAPSettings.defaults['GROUP_MAPPING'] = {}
+    GROUP_CACHE_TIMEOUT = 1200
+
+    @receiver(populate_user)
+    def staff_superuser_populate(sender, user, ldap_user, **kwargs):
+        user.is_superuser = 'superuser' in ldap_user.group_names
+        user.is_staff = 'staff' in ldap_user.group_names
+        user.is_active = 'active' in ldap_user.group_names
+
+    @receiver(populate_user)
+    def manager_country_attribute_populate(
+        sender, user, ldap_user, **kwargs
+    ):
+        try:
+            profile_map = settings.AUTH_LDAP_USER_ATTR_MAP
+        except AttributeError:
+            profile_map = {}
+        if 'manager' in profile_map:
+            if profile_map['manager'] in ldap_user.attrs:
+                manager_ref = ldap_user.attrs[profile_map['manager']][0]
+                # CN=John Smith,OU=TOR,OU=Corp-Users,DC=mydomain,DC=internal
+                # prevent UnicodeEncodeError
+                if not isinstance(manager_ref, unicode):
+                    manager_ref = manager_ref.decode('utf-8')
+                cn = manager_ref.split(',')[0][3:]
+                user.manager = cn
+        # raw value from LDAP is in profile.country for this reason we assign
+        # some correct value
+        user.country = None
+        if 'country' in profile_map:
+            if profile_map['country'] in ldap_user.attrs:
+                country = ldap_user.attrs[profile_map['country']][0]
+                if len(country) == 2:
+                    # assign None if `country` doesn't exist in Country
+                    user.country = getattr(Country, country.lower(), None)
```

Comment:
use `Country.id_from_name(country.lower())`(wrap it with some try except instead of this if condition)